src/facebookanalysis/wc.go

```diff
@@ -49,18 +77,27 @@ func Map(value string) *list.List {
 
 // you need to provide the code to iterate over list and add values
 func Reduce(key string, values *list.List) string { 
-    var total = 0
     
+    t := trie.NewTrie()
+    total := 0
+
     for e := values.Front(); e != nil; e = e.Next() {
-       
-        if val, err := strconv.Atoi(e.Value.(string)); err == nil { 
-            total += val
-        } else {
-            fmt.Printf("Error converting the interface to an integer\n")
-        }
+      
+        total += 1
+        t.Add(e.Value.(string))
+
+    }
+
+
+    ntp := NameTreePair{Total: total, PrefixTree: t.Dump()}
+
+    encoded, err := json.Marshal(ntp)
```

Comment:
Hmm... It think this will answer your question.

The way the Reduce function works in the framework is that it takes as input, a key and a list of values from the key pairs, and returns a single "reduced" string to be associated with that key in the output.

Does that make sense? So what we return is associated with the string `key`.